openmetadata-ui/src/main/resources/ui/src/components/Applications/AppSchedule/AppSchedule.component.tsx

```diff
@@ -26,10 +37,39 @@ const AppSchedule = ({
   appData,
   onSave,
   onDemandTrigger,
+  onDeployTrigger,
 }: AppScheduleProps) => {
   const { t } = useTranslation();
   const [showModal, setShowModal] = useState(false);
   const appRunsHistoryRef = useRef<AppRunsHistoryRef>(null);
+  const [isPipelineDeployed, setIsPipelineDeployed] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const fetchPipelineDetails = useCallback(async () => {
+    setIsLoading(true);
+    try {
+      if (
+        appData.appType === AppType.External &&
+        appData.pipelines &&
+        appData.pipelines.length > 0
+      ) {
+        const fqn = appData.pipelines[0].fullyQualifiedName ?? '';
+        const pipelineData = await getIngestionPipelineByFqn(fqn);
+
+        if (pipelineData) {
+          setIsPipelineDeployed(pipelineData.deployed ?? true);
```

Comment:
Mostly it's for typescript check, because deployed is an optional field.
Also if there is pipelineData then we are sure that it will have deployed variable